### PR TITLE
test: [M3-8155,M3-8154] - Add tests for longview client rename and deletion

### PR DESCRIPTION
**Pull request by AzureLatte**
## Description 📝
Add Cypress integration tests using mock API data to confirm the UI flow when a user renames or deletes a Longview client in Cloud Manager.

## Changes  🔄
- Add new tests in e2e longview.spec.ts
- Add new intercepts functions in longview.ts

## How to test 🧪
```
yarn cy:run -s "cypress/e2e/core/longview/longview.spec.ts”
```

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/cypress/e2e/core/longview/longview.spec.ts [hunk 1]
```diff
@@ -266,4 +269,165 @@ describe('longview', () => {
...
+
+    const newClient: LongviewClient = longviewClientFactory.build({
+      api_key: '01AE82DD-6F99-44F6-95781512B64FFBC3',
+      apps: longviewAppsFactory.build(),
+      created: new Date().toISOString(),
+      id: 338283,
+      install_code: '748632FC-E92B-491F-A29D44019039017C',
+      label: randomLabel(),
+      updated: new Date().toISOString(),
+    });
```
Comment (mjac0bs) [inline packages/manager/cypress/e2e/core/longview/longview.spec.ts:297]:
```suggestion
    const client: LongviewClient = longviewClientFactory.build();

    const newClient: LongviewClient = longviewClientFactory.build({
    ...client, label: randomLabel()
    });
```

If we just want to create a new (renamed) client with the rest of the fields the same, we could save some space with the spread operator.

**Review by mjac0bs — APPROVED** [review]
Thanks @AzureLatte, the test cases covered the expected UI interactions and passed reliably. 

I left a couple of nits about the client objects generated from factories.

File: packages/manager/.changeset/pr-10644-tests-1720540166205.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/manager": Tests
+---
+
+Add tests for longview client rename and deletion ([#10644](https://github.com/linode/manager/pull/10644))
```
[the remark this answers is not in the document]
Comment (AzureLatte) [reply packages/manager/.changeset/pr-10644-tests-1720540166205.md]: Just update, thanks!

File: packages/manager/cypress/e2e/core/longview/longview.spec.ts (hunk 1 above)
Comment (AzureLatte) [reply packages/manager/cypress/e2e/core/longview/longview.spec.ts]: Updated, thank you!

**Review by hana-linode — APPROVED** [review]
Tests passed ✅

**Pull request merged**